### PR TITLE
Replaced PSG for JT89

### PR DESCRIPTION
**Pull request by jotego**
I have written a new PSG (JT89) with correct volume (2dB steps instead of 3dB) and with rounded edges (a fast low pass filter is applied).

The volume correction is very obvious. Now some sounds can be heard in games which were too faint before. The rounded edges may not be heard directly, but it helps the 1-bit DAC as it reduces high frequency content, thus sound should be more clear.

About the mixing between FM and PSG: some games feel like boosting more the PSG in comparison with the FM, but because Sonic uses very loud settings for PSG I have not boosted it more.

JT12 is also updated to correct a tiny 1 LSB error in detuning parameter for some very high settings.

I think this version sounds better overall.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -4,3 +4,6 @@
 [submodule "jt12"]
 	path = jt12
 	url = https://github.com/jotego/jt12
+[submodule "jt89"]
+	path = jt89
+	url = git@github.com:jotego/jt89.git
```
Comment (gyurco) [inline .gitmodules]: Better to have https:// url here, git@github.com is for ssh access, and probably only you have ssh access for your own repo.

**Comment by harbaum** [comment]
What is the scandoubler comment about? I don't see any scandoubler changes.

**Comment by jotego** [comment]
I had actually tried something with the scandoubler and that got committed locally by mistake, so I had to undo the changes. 

There is nothing changed for the scandoubler in the actual merge. This is purely a sound merge request.

**Comment by jotego** [comment]
That demo is haunting me! You're right, it's missing sounds there for some reason. I close the request for now. I will fix that first.

**Comment by gyurco** [comment]
It's your nemesis :) Without that demo I think the core would be considered complete two month ago.

**Comment by harbaum** [comment]
Has anyone ever tried the counseling service they offer in their demo? Sounds like it's exactly what you need :-)

**Comment by gyurco** [comment]
No more things has to be cracked from my side :)

**Pull request closed without merging**